server/src/domain/person/person.service.ts

```diff
@@ -488,19 +488,23 @@ export class PersonService {
       imageHeight,
     } = face;
 
-    const [asset] = await this.assetRepository.getByIds([assetId]);
-    if (!asset?.resizePath) {
+    const asset = await this.assetRepository.getById(assetId, { exifInfo: true });
+    if (!asset?.exifInfo?.exifImageHeight || !asset?.exifInfo?.exifImageWidth) {
       return false;
     }
+
     this.logger.verbose(`Cropping face for person: ${person.id}`);
     const thumbnailPath = StorageCore.getPersonThumbnailPath(person);
     this.storageCore.ensureFolders(thumbnailPath);
 
-    const halfWidth = (x2 - x1) / 2;
-    const halfHeight = (y2 - y1) / 2;
+    const widthScale = asset.exifInfo.exifImageWidth / imageWidth;
+    const heightScale = asset.exifInfo.exifImageHeight / imageHeight;
+
+    const halfWidth = (widthScale * (x2 - x1)) / 2;
+    const halfHeight = (heightScale * (y2 - y1)) / 2;
 
-    const middleX = Math.round(x1 + halfWidth);
-    const middleY = Math.round(y1 + halfHeight);
+    const middleX = Math.round(widthScale * x1 + halfWidth);
+    const middleY = Math.round(heightScale * y1 + halfHeight);
```

Comment:
I don't think if this will always work. We also use `fit: 'outside'`, which basically means there are no guarantees between the original size of the image and the resulting jpeg thumbnail, to which the coordinates for the face belong.

![image](https://github.com/immich-app/immich/assets/4334196/164db1f8-ccb5-46b6-ac2b-3b796bf26ff6)

Reply:
> there are no guarantees between the original size of the image and the resulting jpeg thumbnail, to which the coordinates for the face belong

I don't think this is right. An outer fit just means the smaller dimension is set to 1440 and the larger dimension is scaled based on aspect ratio.

But I think you might be onto something here. All of the pictures in the test dataset I'm working with are horizontal, so this code might not be handling certain aspect ratios correctly.

Reply:
You are trying to reverse map the face coordinates of the jpeg back to the original image, but this can only work if the thumbnail was not cropped at all. If it was I don't think you can reliable reverse-map the coordinates.

Reply:
Hmm I just tried cropping an image in a bunch of ways so it has extreme horizontal or vertical aspect ratios and it still works. Also tried cropping so the face is at the very most corner of the image and that works too. I'm kinda stumped.